qiskit/optimization/converters/inequality_to_equality.py

```diff
@@ -42,26 +42,30 @@ class InequalityToEquality:
 
     _delimiter = '@'  # users are supposed not to use this character in variable names
 
-    def __init__(self) -> None:
+    def __init__(self, name: Optional[str] = None, mode: str = 'auto') -> None:
+        """
+        Args:
+            name: The name of the converted problem. If not provided, the name of the input
+                  problem is used.
+            mode: To chose the type of slack variables. There are 3 options for mode.
+
+                - 'integer': All slack variables will be integer variables.
+                - 'continuous': All slack variables will be continuous variables
+                - 'auto': Try to use integer variables but if it's not possible,
+                   use continuous variables
+        """
         self._src = None  # type: Optional[QuadraticProgram]
         self._dst = None  # type: Optional[QuadraticProgram]
+        self._dst_name = name
+        self._mode = mode
         self._conv = {}  # type: Dict[str, List[Tuple[str, int]]]
         # e.g., self._conv = {'c1': [c1@slack_var]}
 
-    def encode(
-            self, op: QuadraticProgram, name: Optional[str] = None, mode: str = 'auto'
-    ) -> QuadraticProgram:
+    def encode(self, op: QuadraticProgram) -> QuadraticProgram:
```

Comment:
Is the intent to do these changes with no deprecation at all? The release will have been out for a several weeks and users may have code passing parameters on the encode.

Reply:
Yes, It's my intent. Shall I duplicate the `encode` method and add some deprecation message to old one?